subsys/shell/shell.c

```diff
@@ -236,11 +236,11 @@ static bool tab_prepare(const struct shell *shell,
 	(void)shell_make_argv(argc, *argv, shell->ctx->temp_buff,
 			      CONFIG_SHELL_ARGC_MAX);
 
-	if (IS_ENABLED(CONFIG_SHELL_CMDS_SELECT) &&
-	    (strcmp("select", (*argv)[0]) == 0)	 &&
-	    !shell_in_select_mode(shell)) {
-		*argv = *argv + 1;
-		*argc = *argc - 1;
+	if (IS_ENABLED(CONFIG_SHELL_CMDS_SELECT) && (*argc > 0))
+		if ((strcmp("select", (*argv)[0]) == 0)	 &&
```

Comment:
why 2 ifs?

Reply:
Because I cannot guarantee that `(*argc > 0)` will be checked before `strcmp("select", (*argv)[0]) == 0)`

Reply:
Evaluation order is guaranteed to be left-to-right by standard

Reply:
Ok, I've educated myself :) I will fix it

Reply:
fixed